File: Algorithms/0643.maximum-average-subarray-i/maximum-average-subarray-i.go
package problem0643

func findMaxAverage(nums []int, k int) float64 {
	temp := 0
	// 获取第一个temp
	for i := 0; i < k; i++ {
		temp += nums[i]
	}

	max := temp

	for i := k; i < len(nums); i++ {
		// 根据 i 获取 temp
		temp = temp - nums[i-k] + nums[i]

		if max < temp {
			max = temp
		}
	}

	return float64(max) / float64(k)
}

File: Algorithms/0643.maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        """
        @params: nums -> a list of integers; k -> number of integers
        @output: the maximum average of k integers
        """
        avgs = []
        for i in range(len(nums)-k):
            avgs.append(sum(nums[i:i+k]) / k)
        return max(avgs)

    def findMaxAverage1(self, nums: List[int], k: int) -> float:
        temp = 0
        for i in range(k):
            temp += nums[i]

        Max = temp

        for i in range(k,len(nums)):
            temp = temp - nums[i-k] + nums[i]
            if Max < temp:
                Max = temp
        return Max / k
